### ISSUE TITLE
[FR]: ts_version_from using pnpm lock file

### ISSUE THREAD

**Issue by gregmagolan** [aspect-build/rules_ts#308]
### What is the current behavior?

We can only pull the version from the package.json file and the version there must be a pinned version

### Describe the feature

Pulling from the pnpm lock file will allow the package.json specification to be unpinned such as `4.9.x`. We can also pull the integrity out of the lock file so users can easily upgrade to versions of typescript even before they are added to versions.bzl in rules_ts.

### Fund our work

- [ ] Sponsor our open source work by donating a [feature bounty](https://opencollective.com/aspect-build/)

**Comment by alexeagle** [comment]
Same in rulesets where I copied this pattern like rules_swc.

**Comment by Aghassi** [comment]
Might be worth just failing if you don't see a pinned number in the package.json and telling folks how to fix. But lockfile also works. For the record, we pin everything

**Comment by alexeagle** [comment]
Anyone using `npm_translate_lock(name = "npm")` will have a file generated already at `@npm//path/to/pkg:typescript/package.json.bzl` containing:

```
load("@@npm__typescript__4.9.5//:package_json.bzl", _bin = "bin", _bin_factory = "bin_factory")
bin = _bin
bin_factory = _bin_factory
```

So I think we could just change the generation slightly to either write a `version.bzl` file next to this, for every package with a bin, or just export a VERSION symbol from this existing file.

**Comment by alexeagle** [comment]
fixed by #418
